src/Diagnostics.CompilerHost/Controllers/CompilerHostController.cs

```diff
@@ -29,24 +46,33 @@ public async Task<IActionResult> Post([FromBody]JToken jsonBody)
                 return BadRequest("Missing body");
             }
 
-            string script = jsonBody.Value<string>("script");
+            // Get script and reference
+            var script = jsonBody.Value<string>("script");
 
             if (string.IsNullOrWhiteSpace(script))
             {
                 return BadRequest("Missing script from body");
             }
 
-            EntityMetadata metaData = new EntityMetadata(script);
-            CompilerResponse compilerResponse = new CompilerResponse();
-            using (var invoker = new EntityInvoker(metaData, ScriptHelper.GetFrameworkReferences(), ScriptHelper.GetFrameworkImports()))
+            var references = JsonConvert.DeserializeObject<Dictionary<string, string>>(jsonBody["reference"].ToString());
```

Comment:
`jsonBody[reference]` could trigger KeyNotFoundException. Check for this empywhitespace as you are with the script.

Reply:
Can you add an example json that has these references? Just want to know how it looks and what it is.

Reply:
This will not trigger KeyNotFoundException since jsonBody is JObject. I will wrapper this code with try catch because it could fail when deserialize.

Reply:
> Can you add an example json that has these references? Just want to know how it looks and what it is.

The references is actually a dictionary. Client will give init this value in `CompilerHostClient`.